### PR TITLE
search: simplified early return

### PR DESCRIPTION
**Pull request by rvantonder**
Was debugging something related to this function and thing this is slightly simpler?

### CODE REVIEW

File: internal/search/streaming/stream.go [hunk 1]
```diff
@@ -72,6 +72,9 @@ func WithSelect(parent Sender, s filter.SelectPath) Sender {
 	dedup := result.NewDeduper()
 
 	return StreamFunc(func(e SearchEvent) {
+		if parent == nil {
```
Comment (camdencheek) [inline internal/search/streaming/stream.go:75]: I think the idea here was to avoid taking out a lock when streaming is disabled

Comment (rvantonder) [reply internal/search/streaming/stream.go:75]: Hmmm not sure I follow, how does it help in this case?

Anyway, this function is only ever called when `parent` is non-nil: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/graphqlbackend/search_results.go?L888:24&subtree=true. I think it _should_ only be called when it's non-nil, so like, I'd be in favor of actually panic-ing here on non-nil. I realize Go tries to run away from paniccing by just returning stuff but I like the idea of enforcing invariants.

Comment (rvantonder) [reply internal/search/streaming/stream.go:75]: and to be fair the Go compiler has a fair number of panics for around enforcing invariants :P

Comment (camdencheek) [reply internal/search/streaming/stream.go:75]: Oh. Do'h. I read the green and red backwards in the. This is much better -- sorry for the confusion

**Review by camdencheek — APPROVED** [review]

File: internal/search/streaming/stream.go (hunk 1 above)
Comment (camdencheek) [reply internal/search/streaming/stream.go:75]: And I'm totally cool with a panic. I'm a big fan of panics for invariants

**Comment by rvantonder** [comment]
Merging as-is, will think about invariants if/when I get a chance to concentrate more on the stream/batch logic separation.

**Pull request merged**